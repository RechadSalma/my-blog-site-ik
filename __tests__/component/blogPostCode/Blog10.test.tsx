import { render } from "@testing-library/react";
import Blog10Codes from "../../../components/blogPostCodes/Blog10";

describe("<Blog10/> code for blog10 component", () => {
  const blogTenComponent = () => render(<Blog10Codes />);

  test("renders the <code> tag", () => {
    const { getByRole } = blogTenComponent();

    const codeTag = getByRole("code");

    expect(codeTag).toBeInTheDocument();
  });

  // snapshot component
  test("snapshot Blog10 code tag", () => {
    const { container } = blogTenComponent();

    expect(container).toMatchSnapshot();
  });
});
