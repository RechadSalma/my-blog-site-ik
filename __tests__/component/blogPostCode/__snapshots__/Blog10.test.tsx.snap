// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog10/> code for blog10 component snapshot Blog10 code tag 1`] = `
<div>
  <pre
    class="preCode mx-auto"
  >
    <code
      class="codeTag mx-auto"
      role="code"
    >
      <div
        class="comment"
      >
        // Define the object type
      </div>
      <div>
        <span
          class="keyword"
        >
          interface 
        </span>
        object {
      </div>
      <div
        class="tab1"
      >
        <div>
          name: 
          <span
            class="value"
          >
            string | null
          </span>
          ;
        </div>
        <div>
          age: 
          <span
            class="value"
          >
            number | null
          </span>
          ;
        </div>
      </div>
      <div>
        };
      </div>
      <br />
      <div
        class="comment"
      >
        // Define the dictionary object type 
      </div>
      <div>
        <span
          class="keyword"
        >
          interface 
        </span>
        dictionary {
      </div>
      <div
        class="tab1"
      >
        <div>
          [idNumber: 
          <span
            class="value"
          >
            number
          </span>
          ]: object;
        </div>
      </div>
      <div>
        };
      </div>
    </code>
  </pre>
</div>
`;
